backend/src/routes/projectRouter.ts

```diff
@@ -124,6 +124,23 @@ projects.post(
     }
   }
 );
+// Star Project
+projects.post("/v1/projects/:id/star", async (req: Request, res, next) => {
+  await projectController.addStarrer(req.body.user.id, req.body.project);
+});
+
+// Unstar Project
+projects.post("/v1/projects/:id/unstar", async (req: Request, res, next) => {
+  await projectController.removeStarrer(req.body.user.id, req.body.project);
+});
+
+//Get Starred Projects
+projects.get("/v1/projects/get-starred", async (req: Request, res, next) => {
```

Comment:
This endpoint is better named `/v1/users/:id/projects-starred` because then we can look up any user's list of starred projects.

In frontend tho, we'd need the user session's ID, like:
```
function Component() {
  const { user } = useSession();
  ...
  fetch("/api/v1/users/" + user.id + "/projects-starred")
  ...
```